pkg/collector/corechecks/servicediscovery/apm/detect.go

```diff
@@ -231,7 +233,7 @@ func javaDetector(ctx usm.DetectionContext) Instrumentation {
 	}
 	for _, name := range toolOptionEnvs {
 		if val, ok := ctx.Envs.Get(name); ok {
-			if strings.Contains(val, "-javaagent:") && strings.Contains(val, "dd-java-agent.jar") {
+			if javaAgentRegex.MatchString(val) {
```

Comment:
Since here it is all the args and not just a single one, it will match any command line with both `javaagent` and `datadog`, not necessarily only with `datadog` as the java agent.

Reply:
Modified the regex to not allow whitespace characters between the various patterns like `datadog` and the `.jar` suffix. That should match the behaviour we had previously with `strings.Contains`